ui/components/multichain-accounts/smart-contract-account-toggle/smart-contract-account-toggle.test.tsx

```diff
@@ -0,0 +1,239 @@
+
+    it('shows toggle as ON for supported networks', () => {
+      const { container } = render({
+        networkConfig: mockNetworkConfig,
+      });
+
+      const toggle = screen.getByRole('checkbox');
+      expect(toggle.closest('.toggle-button')).toHaveClass('toggle-button--on');
+      expect(container).toMatchSnapshot();
+    });
+
+    it('shows toggle as OFF for unsupported networks', () => {
+      const { container } = render({
+        networkConfig: mockUnsupportedNetworkConfig,
+      });
+
+      const toggle = screen.getByRole('checkbox');
+      expect(toggle.closest('.toggle-button')).toHaveClass(
+        'toggle-button--off',
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('displays different network names correctly', () => {
+      render({
+        networkConfig: mockUnsupportedNetworkConfig,
+      });
+
+      expect(screen.getByText('Sepolia')).toBeInTheDocument();
+    });
+  });
+
+  describe('Toggle Actions', () => {
+    it('downgrades account when turning OFF supported network', async () => {
+      render({
+        networkConfig: mockNetworkConfig, // isSupported: true
+      });
+
+      const toggle = screen.getByRole('checkbox');
+      fireEvent.click(toggle);
+
+      await waitFor(() => {
+        expect(mockDowngradeAccount).toHaveBeenCalledWith(mockAddress);
+      });
+      expect(mockUpgradeAccount).not.toHaveBeenCalled();
+    });
+
+    it('upgrades account when turning ON unsupported network', async () => {
+      render({
+        networkConfig: mockUnsupportedNetworkConfig, // isSupported: false
+      });
+
+      const toggle = screen.getByRole('checkbox');
+      fireEvent.click(toggle);
+
+      await waitFor(() => {
+        expect(mockUpgradeAccount).toHaveBeenCalledWith(
+          mockAddress,
+          mockUnsupportedNetworkConfig.upgradeContractAddress,
+        );
+      });
+      expect(mockDowngradeAccount).not.toHaveBeenCalled();
+    });
+
+    it('prevents upgrade when contract address is missing', async () => {
+      const networkWithoutUpgradeAddress = {
+        ...mockUnsupportedNetworkConfig,
+        upgradeContractAddress: undefined,
+      };
+
+      render({
+        networkConfig: networkWithoutUpgradeAddress,
+      });
+
+      const toggle = screen.getByRole('checkbox');
+      fireEvent.click(toggle);
+
+      await waitFor(() => {
+        expect(mockUpgradeAccount).not.toHaveBeenCalled();
+        expect(mockDowngradeAccount).not.toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('Disabled State', () => {
+    it('disables toggle during pending requests', () => {
+      mockUseBatchAuthorizationRequests.mockReturnValue({
+        hasPendingRequests: true,
+      });
+
+      const { container } = render();
+
+      const toggle = screen.getByRole('checkbox');
+      expect(toggle.closest('.toggle-button')).toHaveClass(
+        'toggle-button--disabled',
+      );
+      expect(container).toMatchSnapshot();
+    });
+
```

Comment:
Nit: We could have added a test to check when `upgradeContractAddress` is not defined, that the toggle is `disabled` (but it would be pretty similar to the other one `'prevents upgrade when contract address is missing'`, so that's probably ok)